central/administration/events/handler/handler_test.go

```diff
@@ -57,7 +58,9 @@ func (s *handlerTestSuite) TestConsumeEvents() {
 		defer s.mutex.RUnlock()
 		return addCalled
 	}
-	s.datastore.EXPECT().AddEvent(s.handler.eventWriteCtx, event).Do(addSetCalledFn)
+	concurrency.WithLock(&s.mutex, func() {
```

Comment:
Do we really need this lock? If we move `s.handler.Start()` after everything is setup (just before `s.eventStream.Produce(event)`) then it should work without it.